src/snowflake/cli/plugins/nativeapp/manager.py

```diff
@@ -564,12 +579,75 @@ def deploy(
             self._apply_package_scripts()
 
             # 3. Upload files from deploy root local folder to the above stage
+            stage_fqn = stage_fqn or self.stage_fqn
             diff = self.sync_deploy_root_with_stage(
                 bundle_map=bundle_map,
                 role=self.package_role,
                 prune=prune,
                 recursive=recursive,
+                stage_fqn=stage_fqn,
                 local_paths_to_sync=local_paths_to_sync,
             )
 
+        if validate:
+            self.validate(use_scratch_stage=False)
+
         return diff
+
+    def validate(self, use_scratch_stage: bool = False):
+        """Validates Native App setup script SQL."""
+        cc.step(f"Validating Snowflake Native App setup script.")
```

Comment:
yeah it could be considered a phase, but the output looks a little wonky since the diff log uses `"\n.join` on a bunch of strings, so they don't come out indented: 
![Uploading image.png…]()

Reply:
The diff output is bad, I have a fix locally that I will submit in the next few days. So it's not a really good comparison.

Reply:
changed it to a phase